pkg/controller/syncmachineset/gcpactuator_test.go

```diff
@@ -181,35 +174,28 @@ func TestGCPAcuator(t *testing.T) {
 			if test.expectedErr {
 				assert.Error(t, err, "expected error for test case")
 			} else {
-				test.validate(t, generatedMachineSets, test.expectedMachineSets)
+				validateMachineSets(t, generatedMachineSets, test.expectedMachineSets)
 			}
 		})
 	}
 }
 
 func validateMachineSets(t *testing.T, mSets []*machineapi.MachineSet, expectedMS []machineSetInfo) {
-	for _, expected := range expectedMS {
-		found := false
-		for _, ms := range mSets {
-			if ms.Name == expected.name {
-				found = true
+	assert.Equal(t, len(expectedMS), len(mSets), "different number of machine sets generated than expected")
 
-				replicas := int(*ms.Spec.Replicas)
-				assert.Equal(t, expected.replicas, replicas, "replica count mismatch")
+	for i, expected := range expectedMS {
+		assert.Equal(t, expected.name, mSets[i].Name, "did not find expected machineset: %s", expected.name)
 
-				gcpProvider, ok := ms.Spec.Template.Spec.ProviderSpec.Value.Object.(*gcpprovider.GCPMachineProviderSpec)
-				assert.True(t, ok, "failed to convert to gcpProviderSpec")
+		assert.Equal(t, expected.replicas, int64(*mSets[i].Spec.Replicas), "replica mismatch")
```

Comment:
That is interesting that replicas in machineset is int32 while replicas in machine pool is int64.

Reply:
interesting indeed [surprised emoji here]